csharp/src/Ice/OutgoingAsync.cs

```diff
@@ -462,6 +562,24 @@ public void Abort(Ice.Exception ex)
             }
         }
 
+        public void Abort(Ice.RemoteException ex)
+        {
+            Debug.Assert(ChildObserver == null);
+            if (ExceptionImpl(ex))
+            {
+                InvokeExceptionAsync();
+            }
+            else if (ex is Ice.CommunicatorDestroyedException)
+            {
+                //
+                // If it's a communicator destroyed exception, swallow
+                // it but instead notify the user thread. Even if no callback
+                // was provided.
+                //
+                throw ex;
+            }
+        }
```

Comment:
I don't think this method is necessary. Abort is only called for the user thread when the invocation is by `OutgoingAsync.Invoke`. It doesn't handle the response from the server.

Reply:
Fixing.